man/man5/zfs-module-parameters.5

```diff
@@ -1752,14 +1769,97 @@ Default value: \fB2\fR.
 .ad
 .RS 12n
 Max concurrent I/Os per top-level vdev (mirrors or raidz arrays) allowed during
-scrub or resilver operations.
+legacy scrub or resilver operations.
 .sp
 Default value: \fB32\fR.
 .RE
 
 .sp
 .ne 2
 .na
+\fBzfs_scan_md_maxinflight\fR (int)
+.ad
+.RS 12n
+Max concurrent I/Os allowed during scrub or resilver metadata scanning phase.
+.sp
+Default value: \fB1024\fR.
```

Comment:
>> When our ~1 second per txg is up

>Not quite sure what you mean by this....

when doing the metadata scanning, dsl_scan_check_suspend() can tell us that we need to pause (finish this txg) after 1-5 seconds (zfs_scrub_min_time_ms to zfs_txg_timeout).

> how long does it take to finish these

I mean if we have 1024 outstanding zio's, and then dsl_scan_async_block_should_pause() returns TRUE (perhaps after 1 second of scrubbing), then the zio_wait() in dsl_scan_sync() has to wait for those 1024 zio's to complete, which could take 10 seconds.  That seems bad.  This is why we have zfs_top_maxinflight, so that we can pause in a reasonable amount of time (50 per disk is no more than ~500ms).